ui-v2/app/adapters/application.js

```diff
@@ -1,10 +1,23 @@
 import Adapter from './http';
 import { inject as service } from '@ember/service';
+import config from 'consul-ui/config/environment';
 
 export const DATACENTER_QUERY_PARAM = 'dc';
+export const NSPACE_QUERY_PARAM = 'ns';
 export default Adapter.extend({
   repo: service('settings'),
+  nspaceRepo: service('repository/nspace/enabled'),
   client: service('client/http'),
+  formatNspace: function(nspace) {
+    if (config.CONSUL_NSPACES_ENABLED) {
+      return nspace !== '' ? { [NSPACE_QUERY_PARAM]: nspace } : undefined;
```

Comment:
does this need to be url encoded? wasn't sure if `request` does that for you or not

Reply:
Yeah `request` encodes every variable you pass inside the template literals, both keys and values.

Reply:
This isn't strictly correct 😁 . Basically its not request that does it its ember-data/jQuery.

When writing the template literal any variables used in the URL itself is encoded. Then any body used in anything other that a GET request is sent as JSON encoded.

A GET request with a body is a little special and follows the old ember-data adapter functionality, in that any body with a GET request gets moved to the GET params (as you can't send a body with a GET request). Under the hood ember-data relied on jQuery to do this and when we did our data layer refactor we tried to replicate this functionality completely, you can see it here with a comment on urlencoding https://github.com/hashicorp/consul/pull/5637/files#diff-d12a13b0f2b1e3556993c370c1ee3ddfR182.

I've just double checked this and it does get url encoded, but we are also protected somewhat by the fact that namespaces can only have DNS compatible names, and therefore probably don't need url encoding anyhow